tka/state.go

```diff
@@ -193,6 +207,32 @@ func (s State) applyVerifiedAUM(update AUM) (State, error) {
 		out.Keys = append(out.Keys[:idx], out.Keys[idx+1:]...)
 		return out, nil
 
+	case AUMAddDenylistNodeKey:
+		for i := range s.BannedNodekeys {
+			if bytes.Equal(update.NodeKey, s.BannedNodekeys[i]) {
+				return State{}, errors.New("entry already exists")
+			}
+		}
+		out := s.cloneForUpdate(&update)
+		// Validity of node-key already checked by StaticValidate in aumVerify().
+		out.BannedNodekeys = append(out.BannedNodekeys, update.NodeKey)
+		return out, nil
+
+	case AUMRemoveDenylistNodeKey:
+		idx := -1
+		for i := range s.BannedNodekeys {
+			if bytes.Equal(update.NodeKey, s.BannedNodekeys[i]) {
+				idx = i
+				break
+			}
+		}
+		if idx < 0 {
+			return State{}, errors.New("no such entry")
+		}
+		out := s.cloneForUpdate(&update)
+		out.BannedNodekeys = append(out.BannedNodekeys[:idx], out.BannedNodekeys[idx+1:]...)
```

Comment:
Would it be better to do the O(1) (but not order-preserving) "swap with the last element" strategy rather than a O(n) shift? I suppose it's already O(n) with the scan, so maybe that doesn't matter.

Reply:
Ooooo, what is the swap with last element strat? 

```go
out.BannedNodeKeys[idx] = out.BannedNodeKeys[len(out.BannedNodeKeys)-1]
out.BannedNodeKeys = out.BannedNodeKeys[:len(out.BannedNodeKeys)-2]
```

or such? does that work for all values of `idx`?

Reply:
Yeah, that's the trick. I think the second line is just `len-1` though.

If `len(out.BannedNodeKeys) == 1` then skip the first assignment (the only element is the one being deleted).